mqtt/mqtt-bridge/src/bridge.rs

```diff
@@ -1,541 +1,126 @@
-                // else it checks that the received topic starts with local prefix and removes the local prefix
-                .map_or(Some(topic_name), |local_prefix| {
-                    let prefix = format!("{}/", local_prefix);
-                    topic_name.strip_prefix(&prefix)
-                })
-                // match topic without local prefix with the topic filter pattern
-                .filter(|stripped_topic| mapper.topic_filter.matches(stripped_topic))
-                .map(|stripped_topic| {
-                    if let Some(remote_prefix) = mapper.topic_settings.remote() {
-                        format!("{}/{}", remote_prefix, stripped_topic)
-                    } else {
-                        stripped_topic.to_string()
-                    }
-                })
-        })
-    }
-}
-
-#[async_trait]
-impl<S> EventHandler for MessageHandler<S>
-where
-    S: StreamWakeableState + Send,
-{
-    type Error = BridgeError;
-
-    async fn handle(&mut self, event: &Event) -> Result<Handled, Self::Error> {
-        if let Event::Publication(publication) = event {
-            let forward_publication =
-                self.transform(&publication.topic_name)
-                    .map(|topic_name| Publication {
-                        topic_name,
-                        qos: publication.qos,
-                        retain: publication.retain,
-                        payload: publication.payload.clone(),
-                    });
-
-            if let Some(publication) = forward_publication {
-                debug!("Save message to store");
-                self.store.push(publication).map_err(BridgeError::Store)?;
-
-                return Ok(Handled::Fully);
-            } else {
-                warn!("No topic matched");
+        match select(local_pump, remote_pump).await {
+            Either::Left(_) => {
+                shutdown_handle.shutdown().await?;
+            }
+            Either::Right(_) => {
+                shutdown_handle.shutdown().await?;
             }
         }
 
-        Ok(Handled::Skipped)
-    }
-}
-
-pub struct UpstreamHandler<S> {
-    messages: MessageHandler<S>,
-    rpc: RpcHandler,
-}
-
-#[async_trait]
-impl<S> EventHandler for UpstreamHandler<S>
-where
-    S: StreamWakeableState + Send,
-{
-    type Error = BridgeError;
-
-    async fn handle(&mut self, event: &Event) -> Result<Handled, Self::Error> {
-        // try to handle as RPC command first
-        if self.rpc.handle(&event).await? == Handled::Fully {
-            return Ok(Handled::Fully);
-        }
-
-        // handle as an event for regular message handler
-        self.messages.handle(&event).await
+        debug!("Bridge {} stopped...", self.connection_settings.name());
+        Ok(())
     }
 }
 
 /// Bridge error.
 #[derive(Debug, thiserror::Error)]
 pub enum BridgeError {
-    #[error("failed to save to store.")]
+    #[error("Failed to save to store.")]
```

Comment:
I see RpcError uses lowercase and I think from other error I saw it was using lowercase